### PR TITLE
Toggle Button: created toggle button

### PR DESCRIPTION
**Pull request by krushnakantanayak**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-62

### Description

- Created Toggle button 

### Screenshots

expensetoggle
incomeToggle

### Tips
```bash
class HomeScreen extends StatefulWidget {
  @override
  State<HomeScreen> createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  bool isIncome = true;

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Scaffold(
      backgroundColor: theme.colors.light,
      body: Center(
        child: MinyToggleButton(
          value: isIncome,
          onChanged: (value) {
            setState(() {
              isIncome = value;
            });
          },
        ),
      ),
    );
  }
}
```

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'Staging' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin
- [x] All hard coded strings are present in constants file

### Primary reviewer checklist

- [x] All items in the author checklist is checked or not applicable
- [x] All files including .lock and assets are reviewed
- [x] Reasonable UI flows/cases related to the change are manually tested
- [x] No commented-out code or unnecessary debug statements.
- [x] No spelling or grammatical errors in code comments or documentation.
- [x] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [x] The branch is up-to-date with the target branch (e.g., staging).
- [x] The code adheres to established coding standards.

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: lib/widgets/buttons/toggle_button.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+
+import '../../theme/theme.dart';
+
+class CreateToggleButton extends StatefulWidget {
```
Comment (quiteachaos) [inline lib/widgets/buttons/toggle_button.dart]: rename class to MinyToggleButton

File: lib/widgets/buttons/toggle_button.dart [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      child: GestureDetector(
+        onTap: () {
+          setState(() {
+            value = !value;
+          });
+          widget.onChanged(value);
+        },
+        child: AnimatedContainer(
+          height: theme.sizing.height.s8,
+          width: theme.sizing.height.s17,
```
Comment (quiteachaos) [inline lib/widgets/buttons/toggle_button.dart]: width: theme.sizing.width.s17,

File: lib/widgets/buttons/toggle_button.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          decoration: BoxDecoration(
+            color: theme.colors.contrastLight,
+            borderRadius: BorderRadius.circular(
+              theme.borderradius.medium,
+            ),
+          ),
+          alignment: value ? Alignment.centerRight : Alignment.centerLeft,
+          child: Container(
+            height: theme.sizing.height.s8,
+            width: theme.sizing.height.s12,
```
Comment (quiteachaos) [inline lib/widgets/buttons/toggle_button.dart]: width: theme.sizing.width.s12,

File: lib/widgets/buttons/toggle_button.dart (hunk 1 above)
Comment (krushnakantanayak) [reply lib/widgets/buttons/toggle_button.dart]: Rename done

File: lib/widgets/buttons/toggle_button.dart (hunk 2 above)
Comment (krushnakantanayak) [reply lib/widgets/buttons/toggle_button.dart]: corrected

File: lib/widgets/buttons/toggle_button.dart (hunk 3 above)
Comment (krushnakantanayak) [reply lib/widgets/buttons/toggle_button.dart]: corrected

**Review by quiteachaos — APPROVED** [review]

**Review by satyabratanayak — APPROVED** [review]

**Pull request merged**